File: Python/Searching/binary-search.py
def binary_search(arr, target):
    low = 0
    high = len(arr) - 1

    while low <= high:
        mid = (low + high) // 2

        if arr[mid] < target:
            low = mid + 1
        elif arr[mid] > target:
            high = mid - 1
        else:
            return mid
    return -1


# Testing
arr = [1, 2, 3, 4, 5, 6, 7, 8, 9]
target = 5

result = binary_search(arr, target)

if result != -1:
    print("Element:{0} is present at index {1}".format(target, result))
else:
    print("Element:{0} is not present in array".format(target))

File: JavaScript/Searching/binary-search.js
function binary_search(arr, target) {
    let left = 0;
    let right = arr.length - 1;

    while (left <= right) {
        let mid = Math.floor((left + right) / 2);

        if (arr[mid] === target) {
            return mid;
        } else if (arr[mid] < target) {
            left = mid + 1;
        } else {
            right = mid - 1;
        }
    }

    return -1;
}

// Testing
let arr = [1, 2, 3, 4, 5, 6, 7, 8, 9];
target = 5;

if (binary_search(arr, target) === -1) {
    console.log('Not found');
} else {
    console.log('Found ' + target + ' at index ' + binary_search(arr, target));
}
